src/Microsoft.Diagnostics.Runtime/src/Common/ModuleInfo.cs

```diff
@@ -115,40 +116,38 @@ public VersionInfo Version
         {
             get
             {
-                if (_version is VersionInfo version)
+                if (_version.HasValue)
                 {
-                    return version;
+                    return _version.Value;
```

Comment:
#396

Reply:
hmm.  I know I merged that change but I'm rethinking that.  Overall I care more about readability/maintainability over micro optimizations.  (Though if something actually causes an actual perf issue in a hot loop I'm happy to optimize.)  In this case, this is a jit bug?  It's slow now, but at some point the JIT guys will fix the problem and the HasValue/Value will work as fast as other patterns.

Using `.HasValue` then `.Value` has symmetry and is much clearer in intent than "if (foo is Bar bar)" or `GetValueOrDefault`.  Additionally, `.Value` is saying "I really know this has a value or it'd thrown an exception this method isn't supposed to", versus `.GetValueOrDefault` could silently return the wrong value if the code above it isn't correct (or gets refactored incorrectly).

Essentially, I don't want to litter the code with micro optimizations which make the code harder to read or reason about.  If they have a noticeable perf impact or are in a tight loop that's when I want to use micro-optimizations.